cmd/thv-operator/controllers/mcpserver_kagent.go

```diff
@@ -0,0 +1,202 @@
+		if err := r.Update(ctx, existing); err != nil {
+			return fmt.Errorf("failed to update kagent ToolServer: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// deleteKagentToolServer deletes the kagent ToolServer if it exists
+func (r *MCPServerReconciler) deleteKagentToolServer(ctx context.Context, mcpServer *mcpv1alpha1.MCPServer) error {
+	logger := log.FromContext(ctx)
+
+	kagentToolServer := &unstructured.Unstructured{}
+	kagentToolServer.SetGroupVersionKind(kagentToolServerGVK)
+	kagentToolServer.SetName(fmt.Sprintf("toolhive-%s", mcpServer.Name))
+	kagentToolServer.SetNamespace(mcpServer.Namespace)
+
+	err := r.Get(ctx, types.NamespacedName{
+		Name:      kagentToolServer.GetName(),
+		Namespace: kagentToolServer.GetNamespace(),
+	}, kagentToolServer)
+
+	if errors.IsNotFound(err) {
+		// Already deleted
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("failed to get kagent ToolServer for deletion: %w", err)
+	}
+
+	logger.Info("Deleting kagent ToolServer",
+		"name", kagentToolServer.GetName(),
+		"namespace", kagentToolServer.GetNamespace())
+	if err := r.Delete(ctx, kagentToolServer); err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("failed to delete kagent ToolServer: %w", err)
+	}
+
+	return nil
+}
+
+// createKagentToolServerObject creates an unstructured kagent ToolServer object
+func (*MCPServerReconciler) createKagentToolServerObject(mcpServer *mcpv1alpha1.MCPServer) *unstructured.Unstructured {
+	kagentToolServer := &unstructured.Unstructured{}
+	kagentToolServer.SetGroupVersionKind(kagentToolServerGVK)
+	kagentToolServer.SetName(fmt.Sprintf("toolhive-%s", mcpServer.Name))
+	kagentToolServer.SetNamespace(mcpServer.Namespace)
+
+	// Build the service URL for the ToolHive MCP server
+	serviceName := createServiceName(mcpServer.Name)
+	serviceURL := fmt.Sprintf("http://%s.%s.svc.cluster.local:%d",
+		serviceName, mcpServer.Namespace, mcpServer.Spec.Port)
+
+	// Determine the config type based on ToolHive transport
+	var configType string
+	var config map[string]interface{}
+
+	switch mcpServer.Spec.Transport {
+	case "sse":
+		configType = kagentConfigTypeSSE
+		config = map[string]interface{}{
+			kagentConfigTypeSSE: map[string]interface{}{
+				"url": serviceURL,
+			},
+		}
+	case "streamable-http":
+		configType = kagentConfigTypeStreamableHTTP
+		config = map[string]interface{}{
+			kagentConfigTypeStreamableHTTP: map[string]interface{}{
+				"url": serviceURL,
+			},
+		}
+	default:
+		// For stdio or any other transport, default to SSE
+		// since ToolHive exposes everything via HTTP
+		configType = kagentConfigTypeSSE
+		config = map[string]interface{}{
+			kagentConfigTypeSSE: map[string]interface{}{
+				"url": serviceURL,
+			},
+		}
+	}
+
+	config["type"] = configType
+
+	// Build the spec
+	spec := map[string]interface{}{
+		"description": fmt.Sprintf("ToolHive MCP Server: %s", mcpServer.Name),
+		"config":      config,
+	}
+
+	kagentToolServer.Object = map[string]interface{}{
```

Comment:
Is it possible to use the object here https://github.com/kagent-dev/kagent/blob/main/go/controller/api/v1alpha2/agent_types.go ?